### DISCUSSION TITLE
How does exception handling on other architectures (e.g. ARM) differ from x86?

### DISCUSSION THREAD

**Discussion by kennystrawnmusic** [phil-opp/blog_os#1136] [Questions] [answered]
Just need to get an idea for how to go about handling exceptions on foreign CPUs in the event that it becomes necessary to port to them, since as far as I'm aware there's no, for example, `extern "cortex-m-interrupt"` ABI to work off of. Not an urgent matter, just a curiosity.

**Comment by bjorn3** [comment, marked as answer]
If there is no dedicated call conv implemented in rustc, you will have to write some `global_asm!()` to translate from the platform interrupt call conv to the C call conv and then call an `extern "C"` function with the rust code for the interrupt handler.
